### PR TITLE
Fix search results ranking and grouping

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

Shout out @markerikson for flagging the search issue and @shortcuts for the help fixing it!

Makes a few improvements to searching, some of which are applied to the search index already outside of this PR:

- Fixes ranking so that React APIs and Learn are ranked at the top again. I think this regressed at some point. The fix was to move the `pageRank` attribute above `exact` in the sorting and ranking settings.
- Split up the "React APIs" into different groups:
  - React APIs
  - React DOM APIs
  - React Server Components
  - Rules of React
  - Legacy React APIs (downranked)

The changes included here are:
- Display the groups again (previously it was just a flat list with no grouping headings like React APIs)
- Display more results (previously there were a very small number of results)

To do that I updated DocSearch depts, and added/updated the `searchParameters`.

## Before

Screenshot 2024-09-23 at 3 35 08 PM

## After

Screenshot 2024-09-23 at 3 32 01 PM

### CODE REVIEW

**Comment by markerikson** [comment]
Looks great!

File: package-lock.json
Comment (poteto) [inline package-lock.json:1]: I think this might have been accidentally added?

**Review by shortcuts — COMMENTED** [review]
looks perfect! the bug is being fixed on our side too :)

File: package-lock.json
Comment (rickhanlonii) [reply package-lock.json:1]: oh snap fixed here https://github.com/reactjs/react.dev/pull/7184

**Pull request merged**